lms/services/course.py

```diff
@@ -131,23 +136,25 @@ def search(  # noqa: PLR0913, PLR0917
             name=name,
             limit=limit,
             organization_ids=organization_ids,
-            h_userid=h_userid,
+            h_userids=h_userids,
         ).all()
 
     def get_courses(
         self,
-        h_userid: str | None,
+        instructor_h_userid: str | None,
         organization: Organization | None = None,
+        h_userids: list[str] | None = None,
     ) -> Select[tuple[Course]]:
         """Get a list of unique courses.
 
         :param organization: organization the courses belong to.
-        :param h_userid: only courses this user has access to.
+        :param instructor_h_userid: return only courses where instructor_h_userid is an instructor.
```

Comment:
As for other entities we have two different parameters here, one to fetch the elements **we** have access to (instructor_h_userid) and one for filtering down by user (h_userids).